diam/reflect_test.go

```diff
@@ -323,6 +323,48 @@ func TestUnmarshalCER(t *testing.T) {
 	}
 }
 
+func TestEmbeddedStruct(t *testing.T) {
```

Comment:
Could you add a test for non struct embedded field as well? Maybe something like this:

`func TestEmbeddedStruct(t *testing.T) {
	msg, err := ReadMessage(bytes.NewReader(testMessage), dict.Default)
	if err != nil {
		t.Fatal(err)
	}
	type Common struct {
		OriginHost  string `avp:"Origin-Host"`
		OriginRealm string `avp:"Origin-Realm"`
	}
	type CER struct {
		Common
		HostIP      net.IP `avp:"Host-IP-Address"`
	}
	var d CER
	if err := msg.Unmarshal(&d); err != nil {
		t.Fatal(err)
	}
	if d.OriginHost != "test" {
		t.Fatal("Embedded struct was not unmarshalled")
	}
	newMsg := Message{
		Header: &Header{
			ApplicationID: 0,
		},
		dictionary: dict.Default,
	}
	err = newMsg.Marshal(&d)
	if err != nil {
		t.Fatal("Failed to marshal struct")
	}
	var readBack CER
	if err = newMsg.Unmarshal(&readBack); err != nil {
		t.Fatal(err)
	}
	if readBack.OriginHost != d.OriginHost {
		t.Fatal("name does not match when mashal/unmarshalling")
	}
	
	// Embedded eith non-struct type
	type CEREmb struct {
		Common
		net.IP `avp:"Host-IP-Address"`
	}
	var e CEREmb
	if err := msg.Unmarshal(&e); err != nil {
		t.Fatal(err)
	}
	if e.OriginHost != "test" {
		t.Fatal("Embedded struct was not unmarshalled")
	}
	expectedIp := net.ParseIP("10.1.0.1")
	if e.IP.String() != expectedIp.String() {
		t.Fatalf("Embedded IP was not unmarshalled: %v != %v", e.IP, expectedIp)
	}
	newMsg = Message{
		Header: &Header{
			ApplicationID: 0,
		},
		dictionary: dict.Default,
	}
	err = newMsg.Marshal(&e)
	if err != nil {
		t.Fatal("Failed to marshal struct")
	}
	var readBackEmb CEREmb
	if err = newMsg.Unmarshal(&readBackEmb); err != nil {
		t.Fatal(err)
	}
	if readBackEmb.OriginHost != e.OriginHost {
		t.Fatal("name does not match when mashal/unmarshalling")
	}
	if readBackEmb.IP.String() != expectedIp.String() {
		t.Fatalf("Host IP does not match when mashal/unmarshalling: %v != %v", readBackEmb.IP, expectedIp)
	}
}
`

Reply:
Fixed in 061a6ec780e303403aa849cf71e3dc3762258b5d